pandas/tests/reshape/merge/test_merge.py

```diff
@@ -2598,3 +2598,31 @@ def test_merge_outer_with_NaN(dtype):
         dtype=dtype,
     )
     tm.assert_frame_equal(result, expected)
+
+
+@pytest.mark.parametrize("string_dtype", tm.STRING_DTYPES)
+def test_merge_ea_with_string(join_type, string_dtype):
```

Comment:
Your test does not cover the copy, works even without.

Reply:
It looks difficult to do. The `self.left` (and `self.right`) are internal objects of the `_MergeOperation` class... 

I can ensure that `df1` and `df2` dtypes have not changed after the call to `merge` in my test but, even without the `copy`, it continues to work. What would you propose ?

Reply:
Switching df1 and df2 in the merge call